pandas/core/_numba/kernels/mean_.py

```diff
@@ -0,0 +1,119 @@
+"""
+Numba 1D aggregation kernels that can be shared by
+* Dataframe / Series
+* groupby
+* rolling / expanding
+
+Mirrors pandas/_libs/window/aggregation.pyx
+"""
+from __future__ import annotations
+
+import numba
+import numpy as np
+
+
+@numba.jit(nopython=True, nogil=True, parallel=False)
+def is_monotonic_increasing(bounds: np.ndarray) -> bool:
+    n = len(bounds)
+    if n == 1:
+        return bounds[0] == bounds[0]
```

Comment:
Is this to stop single element `NaN` sequences from being monotonic increasing?

Reply:
I believe so, yes. This snippet was taken from translating this function, but I was able to remove this condition since we know the inputs should be `int64`s will no `NaN`s

https://github.com/pandas-dev/pandas/blob/ae049aea7b83a63dddad42fae486e44e827025be/pandas/_libs/algos.pyx#L792